File: materials/emissive.py
from materials.material import *

class Emissive(Material):
    def __init__(self, color=Color(1, 1, 1), intensity=5):
        super().__init__(color)
        self.color *= intensity

    def bounce(self, r, normal, hit_pos):
        return None

    def emitted(self):
        return self.color

File: materials/emissive.cpp
#include "emissive.h"

Emissive::Emissive(const Vector &color, const double intensity)
    : Material(color), intensity(intensity) {}

Ray Emissive::bounce(const Ray &r, const Vector &normal,
                     const Vector &hitPos) const {
    return Ray(Vector(), Vector());
}

Vector Emissive::emitted() const {
    return color * intensity;
}